include/manif/impl/se3/SE3_base.h

```diff
@@ -206,6 +231,43 @@ SE3Base<_Derived>::translation() const
   return coeffs().template head<3>();
 }
 
+template <typename _Derived>
+void SE3Base<_Derived>::quat(const SE3Base<_Derived>::QuaternionDataType& quaternion)
+{
+  using std::abs;
+  MANIF_ASSERT(abs(quaternion.template norm()-Scalar(1)) <
+               Constants<Scalar>::eps_s,
+               "The quaternion is not normalized !",
+               invalid_argument);
+
+  asSO3().coeffs() = quaternion.coeffs();
```

Comment:
How about forwarding to `quat(Eigen::MatrixBase)` to avoid code duplication?
```suggestion
quat(quaternion.coeffs());
```